scripts/deploy.ts

```diff
@@ -1,19 +1,211 @@
+     * e.g. if it is the USDC address, users will deposit USDC into the market.
+     */
+    const settlementTokenAddress = networkConstants.usdc // USDC
+    /**
+     * This is the `IOracleWrapper` implementation which gives the price of ETH in the settlement token.
+     * e.g. if the settlement token is a USD stablecoin, then settlementEthOracle should be an ETH/USD oracle wrapper.
+     * If you are deploying a market with a settlement token for which there does not exist an oracle wrapper implementation,
+     * you will need to deploy your own. This script assumes you already have one.
+     */
+    const settlementEthOracleAddress = networkConstants.ethUsdcOracleWrapper // ETH/USD oracleWrapper
+    /**
+     * The fee controller is the address that can change the minting and burning fees.
+     * It is highly recommended that this is set to the Tracer dev multisig (0x0f78e8...), otherwise it will likely not
+     * be considered a trustworthy market.
+     * The feeController can not change annual protocol fee, only the minting and burning fee.
+     */
+    const feeController = networkConstants.devMultisig
+    /**
+     * This should be of the format BASE/QUOTE+SETTLEMENT.
+     * For example, an ETH/BTC market settled in USDC would be called ETH/BTC+USDC.
+     * The factory will then prepend the leverage as necessary.
+     */
+    const poolName = "BTC/USD+USDC"
+
+    /**
+     * Set this to the address of the underlying price feed.
+     * Currently, we only support chainlink oracles. It is possible to implement an oracle wrapper and
+     * a corresponding SMAOracle for other oracle providers, put unless you know what you are doing,
+     * it is recommended to stick with Chainlink for now.
+     * Find the chainlink arbitrum mainnet addresses here: https://data.chain.link/arbitrum/mainnet
+     */
+    const underlyingPriceFeed = networkConstants.ethUsdChainlinkFeed
+
+    /**
+     * Do you want to deploy an SMA (simple moving average) market? This helps mitigate against volatility decay
+     * but also requires a long frontrunning interval, proportional to the length of the SMA sampling period.
+     * e.g. 8 hours of SMA sampling usually requires an 8 hours frontrunning interval.
+     */
+    const usingSMA: boolean = true
+
+    /**
+     * Deploy the oracle wrapper that can either be used as the market oracle, or as the price feed for the market's SMA oracle.
+     */
+    const oracleWrapperFactory = await ethers.getContractFactory(
+        "ChainlinkOracleWrapper"
+    )
+    let oracleWrapper = await oracleWrapperFactory.deploy(
+        underlyingPriceFeed,
+        signers[0].address
+    )
+    console.log("Deployed oracleWrapper: %s", oracleWrapper.address)
+
+    if (usingSMA) {
+        /**
+         * The number of SMA periods refers to how many periods, each equal to updateInterval in length,
+         * the oracle samples from to calculate the moving average.
+         */
+        const numSmaPeriods = 8
```

Comment:
should we allow SMA periods to be adjustable as well instead of hardcoding it?

Reply:
I don't see why not. We don't want to enforce SMA at all in the first place, so we can definitely allow market deployers to choose SMA parameters too IMO.